pkg/tenant/db/errors.go

```diff
@@ -49,22 +54,44 @@ var (
 	ErrTopicNameTooLong          = errors.New("topic name is too long")
 )
 
+// ValidationError represents a validation error for a specific field and may contain
+// an index if the field is a slice.
 type ValidationError struct {
-	err error
+	Field string
+	Err   error
+	Index int
 }
 
-func invalid(err error) *ValidationError {
-	return &ValidationError{err}
+func validationError(field string, err error) *ValidationError {
+	return &ValidationError{
+		Field: field,
+		Err:   err,
+	}
+}
+
+func (e *ValidationError) AtIndex(index int) *ValidationError {
+	e.Index = index
+	return e
 }
 
 func (e *ValidationError) Error() string {
-	return fmt.Sprintf("validation error: %s", e.err)
+	return fmt.Sprintf("invalid field '%s': %s", e.Field, e.Err)
 }
 
 func (e *ValidationError) Is(target error) bool {
-	return errors.Is(e.err, target)
+	return errors.Is(e.Err, target)
 }
 
 func (e *ValidationError) Unwrap() error {
-	return e.err
+	return e.Err
+}
+
+type ValidationErrors []*ValidationError
+
+func (v ValidationErrors) Error() string {
+	errs := make([]string, 0, len(v))
+	for _, e := range v {
+		errs = append(errs, e.Error())
+	}
+	return fmt.Sprintf("%d validation errors occurred:\n%s", len(v), strings.Join(errs, "\n"))
```

Comment:
```suggestion
	return fmt.Sprintf("%d validation errors occurred:\n  -%s", len(v), strings.Join(errs, "\n  -"))
```

A bit nicer formatting suggestion